### PR TITLE
issue_252:sparkles: introducing dynamic rendering of the footer

### PR DESCRIPTION
**Pull request by Beulah-Matt**
# Description

Making the router dynamic using ngSwitch.

To achieve this, we needed to:

```
Render a reusable component in each switchCase 
```

Fixes # (issue) - https://github.com/italanta/elewa-group/issues/252
## Type of change

Please delete options that are not relevant.

# Screenshot (optional)

# Checklist:
- [ ✔️] My code follows the style guidelines of this project
- [ ✔️] I have performed a self-review of my code
- [✔️ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ✔️] My changes generate no new warnings
- [ ✔️] I have added tests that prove my fix is effective or that my feature works
- [✔️ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.html [hunk 1]
```diff
@@ -7,13 +7,26 @@
...
-        <h4>do you want to be part of our mission?</h4>
-        <div class="mission">
-          <elewa-group-button-with-animation [message]="'Become an investor'" [mode]="'dark'" [action]="'/'"></elewa-group-button-with-animation>
-        </div>
-      </div>
+    <elewa-group-footer [ngSwitch]="currentPage">
+      <ng-container *ngSwitchCase="'home'">
+        <elewa-group-footer></elewa-group-footer>
+      </ng-container>
+
```
Comment (IanOdhiambo9) [inline libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.html:14]: You've nested the footer

File: libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.html [hunk 2]
```diff
@@ -7,13 +7,26 @@
     <router-outlet></router-outlet>
   </div>
   <div class="footer">
-    <elewa-group-footer>
-      <div footer class="footer-content">
-        <h4>do you want to be part of our mission?</h4>
-        <div class="mission">
```
Comment (IanOdhiambo9) [inline libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.html:13]: switch needs to happen inside this div

File: libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.ts [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+
+
+ constructor(private route: ActivatedRoute) {}
+
+
+ ngOnInit() {
+   this.route.url.subscribe(urlSegments => {
+           // Update currentPage based on current route
+           this.currentPage = urlSegments[0].path;
+   })
```
Comment (IanOdhiambo9) [inline libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.ts:21]: Looks good!

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]

File: libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.html (hunk 1 above)
Comment (Beulah-Matt) [reply libs/elements/layout/src/lib/pages/elewa-group-main-page/elewa-group-main-page.component.html:14]: @IanOdhiambo9 please review the changes I have made.

**Comment by IanOdhiambo9** [comment]
This issue has been closed

**Pull request closed without merging**